apps/prairielearn/src/question-servers/calculation-subprocess.js

```diff
@@ -47,7 +58,15 @@ function getQuestionRuntimePath(questionServerPath, courseHostPath, courseRuntim
   }
 }
 
+/**
+ * @param {string} func
+ * @param {import('../lib/db-types.js').Course} question_course
+ * @param {import('../lib/db-types.js').Question} question
+ * @param {any} inputData
+ */
```

Comment:
We can't reasonably type the return type of this until we can use actual generics in TypeScript. Even then, we're really just going to be lying to the type checker, but that's safe; the underlying functions in `calculation-worker.js` *are* typed for real.

Reply:
Sounds good.